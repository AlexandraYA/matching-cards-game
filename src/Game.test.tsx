import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import GamePage from './game/gamePage';

describe('renders start page level 1', () => {

  it('should be in page', async () => {
    render(<GamePage />);

    expect(screen.getByText(/Уровень 1/i)).toBeInTheDocument();
    expect(screen.getByText(/попыток 20/i)).toBeInTheDocument();
    expect(screen.getAllByTestId(/card-closed/i)).toHaveLength(12);
    expect(screen.getByTestId("field")).toHaveClass("level1");
  });

});

describe('test click events', () => {

  it('flip card by click', async () => {
    render(<GamePage />);
  
    const field = screen.getByTestId("field");
    fireEvent.click(field.childNodes[0]);
  
    await screen.findByTestId("card-open");
  });
  
  
  it('should not mark as found', async () => {
    render(<GamePage />);
    
    const firstCard = screen.getAllByTestId(/card-closed/i)[0];
    userEvent.click(firstCard);
    expect(firstCard).toHaveClass("open");
    userEvent.click(firstCard);
    expect(firstCard).not.toHaveClass("found");
  });


  it('should not flip back', async () => {
    render(<GamePage />);
    
    const firstCard = screen.getAllByTestId("card-closed")[0];
    userEvent.click(firstCard);
    userEvent.click(screen.getAllByTestId("card-closed")[1]);
    expect(screen.getAllByTestId("card-open")).toHaveLength(2);

    userEvent.click(firstCard);
    expect(screen.getAllByTestId("card-open")).toHaveLength(2);
  });

  it('should not decrease steps', async () => {
    render(<GamePage />);
    
    const firstCard = screen.getAllByTestId("card-closed")[0];
    userEvent.click(firstCard);
    userEvent.click(firstCard);
    userEvent.click(firstCard);
    expect(screen.getByText(/попыток 20/i)).toBeInTheDocument();
  });
});

