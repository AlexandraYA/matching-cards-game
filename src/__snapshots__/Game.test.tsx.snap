// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders start page level 1 should match snapshot 1`] = `
<div
  class="container"
  data-testid="container"
>
  <header
    class="header"
  >
    <h1
      class="main-title"
    >
      Найди все совпадения
    </h1>
    <h3
      class="rules"
    >
      <span>
        Уровень 
        1
      </span>
      <span>
        Шаг 
        20
      </span>
    </h3>
  </header>
  <div
    class="playground"
  >
    <div
      class="coverage"
      data-testid="coverage"
    >
      <button
        class="button"
      >
        Играть
      </button>
    </div>
    <div
      class="field level1"
      data-testid="field"
    >
      <div
        class=""
        data-testid="card-closed"
      >
        <img
          alt="card face 26"
          class="image-face"
          src="monkey.png"
        />
        <img
          alt="card back"
          class="image-back"
          src="card-back-black.png"
        />
      </div>
      <div
        class=""
        data-testid="card-closed"
      >
        <img
          alt="card face 19"
          class="image-face"
          src="fisher.jpg"
        />
        <img
          alt="card back"
          class="image-back"
          src="card-back-black.png"
        />
      </div>
      <div
        class=""
        data-testid="card-closed"
      >
        <img
          alt="card face 23"
          class="image-face"
          src="horse.png"
        />
        <img
          alt="card back"
          class="image-back"
          src="card-back-black.png"
        />
      </div>
      <div
        class=""
        data-testid="card-closed"
      >
        <img
          alt="card face 23"
          class="image-face"
          src="horse.png"
        />
        <img
          alt="card back"
          class="image-back"
          src="card-back-black.png"
        />
      </div>
      <div
        class=""
        data-testid="card-closed"
      >
        <img
          alt="card face 18"
          class="image-face"
          src="duck.png"
        />
        <img
          alt="card back"
          class="image-back"
          src="card-back-black.png"
        />
      </div>
      <div
        class=""
        data-testid="card-closed"
      >
        <img
          alt="card face 28"
          class="image-face"
          src="penguin.png"
        />
        <img
          alt="card back"
          class="image-back"
          src="card-back-black.png"
        />
      </div>
      <div
        class=""
        data-testid="card-closed"
      >
        <img
          alt="card face 28"
          class="image-face"
          src="penguin.png"
        />
        <img
          alt="card back"
          class="image-back"
          src="card-back-black.png"
        />
      </div>
      <div
        class=""
        data-testid="card-closed"
      >
        <img
          alt="card face 11"
          class="image-face"
          src="zebra.png"
        />
        <img
          alt="card back"
          class="image-back"
          src="card-back-black.png"
        />
      </div>
      <div
        class=""
        data-testid="card-closed"
      >
        <img
          alt="card face 18"
          class="image-face"
          src="duck.png"
        />
        <img
          alt="card back"
          class="image-back"
          src="card-back-black.png"
        />
      </div>
      <div
        class=""
        data-testid="card-closed"
      >
        <img
          alt="card face 26"
          class="image-face"
          src="monkey.png"
        />
        <img
          alt="card back"
          class="image-back"
          src="card-back-black.png"
        />
      </div>
      <div
        class=""
        data-testid="card-closed"
      >
        <img
          alt="card face 11"
          class="image-face"
          src="zebra.png"
        />
        <img
          alt="card back"
          class="image-back"
          src="card-back-black.png"
        />
      </div>
      <div
        class=""
        data-testid="card-closed"
      >
        <img
          alt="card face 19"
          class="image-face"
          src="fisher.jpg"
        />
        <img
          alt="card back"
          class="image-back"
          src="card-back-black.png"
        />
      </div>
    </div>
  </div>
  <footer
    class="footer"
  >
    <a
      href="https://www.flaticon.com/ru/free-stickers/"
      rel="noreferrer"
      target="_blank"
      title="свинья стикеры"
    >
      Стикеры от Stickers - Flaticon
    </a>
  </footer>
</div>
`;
